diff -urN iftop-0.13.org/configure.in iftop-0.13/configure.in
--- iftop-0.13.org/configure.in	Wed Jun 11 19:18:36 2003
+++ iftop-0.13/configure.in	Thu Jun 12 09:45:28 2003
@@ -298,7 +298,7 @@
 for curseslib in curses ncurses ; do
     LIBS="$oldLIBS -l$curseslib"
     AC_TRY_LINK([
-#include <curses.h>
+#include <ncurses/curses.h>
         ], [
         mvchgat(0, 0, 1, A_REVERSE, 0, NULL)
         ], [
